ios/FluentUI/Controls/IndeterminateProgressBarView.swift

```diff
@@ -0,0 +1,143 @@
+    }
+
+    public override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        if #available(iOS 13, *) {
+            if previousTraitCollection?.userInterfaceStyle != traitCollection.userInterfaceStyle {
+                updateColors()
+            }
+        }
+    }
+
+    private func addGradientLayer() {
+        gradientLayer.removeFromSuperlayer()
+        gradientLayer.frame = CGRect(x: 0, y: 0, width: frame.width / 2, height: frame.height)
+        gradientLayer.startPoint = CGPoint(x: 0.0, y: 1.0)
+        gradientLayer.endPoint = CGPoint(x: 1.0, y: 1.0)
+        gradientLayer.locations = [0.0, 0.5, 1.0]
+        layer.addSublayer(gradientLayer)
+    }
+
+    private func addAnimation() -> CAAnimationGroup {
+        let groupAnimation = CAAnimationGroup()
+
+        let position = CABasicAnimation(keyPath: "position.x")
+        position.fromValue = -frame.width / 2
+        position.toValue = frame.width * 1.5
+
+        let opacity = CAKeyframeAnimation(keyPath: "opacity")
+        opacity.keyTimes = Constants.opacityKeyFrame
+        opacity.values = Constants.opacityValues
+        opacity.isRemovedOnCompletion = false
+        opacity.fillMode = CAMediaTimingFillMode.forwards
+
+        groupAnimation.animations = [opacity, position]
+        groupAnimation.repeatCount = .infinity
+        groupAnimation.duration = Constants.animationDuration
+        groupAnimation.timingFunction = Constants.animationTimingFunction
+        return groupAnimation
+    }
+
+    private func updateColors() {
+        if let window = window {
+            let appColor = Colors.primary(for: window)
+            let edgeColor = appColor.withAlphaComponent(0).cgColor
+            let centerColor = appColor.cgColor
+
+            gradientLayer.colors = [edgeColor, centerColor, edgeColor]
+            trackLayer.backgroundColor = Colors.Progress.trackTint.cgColor
+        }
+    }
+
+    private struct Constants {
+        static let animationDuration: TimeInterval = 2.5
+        static let animationKey: String = "progressAnimation"
+        static let animationTimingFunction = CAMediaTimingFunction(controlPoints: 0.28, 0.08, 0.72, 0.92)
+        static let opacityKeyFrame: [NSNumber] = [0, 0.375, 0.625, 1]
+        static let opacityValues: [NSNumber] = [0.0, 1.0, 1.0, 0.0]
+        static let progressBarHeight: CGFloat = 2
+    }
+
+    /// progressbar's track
+    private var trackLayer = CALayer()
+
+    /// gradient layer that slides across the track
+    private var gradientLayer = CAGradientLayer()
+
+    /// The grouped animations that will be applied to the gradient layer
+    private var animationGroup = CAAnimationGroup()
+
+    // Don't modify this directly. Instead, call `startAnimating` and `stopAnimating`
+    @objc(isAnimating) public private(set) var isAnimating: Bool = false
```

Comment:
Should move public property higher in the file per https://microsoft.github.io/swift-guide/FileContentOrdering.html